### PR TITLE
[WIP] add a feed system

### PR DESCRIPTION
**Pull request by hbina**
Attempts to solve #51

### CODE REVIEW

**Comment by hbina** [comment]
It currently does not even compile, I have no idea how to proceed past this wall of error messages.

```
error[E0599]: no method named `join` found for type `(&specs::storage::Storage<'_, components::combat::Damage, shred::res::Fetch<'_, specs::storage::MaskedStorage<components::combat::Damage>>>, specs::storage::AntiStorage<'_>, &specs::storage::Storage<'_, components::combat::Speed, shred::res::Fetch<'_, specs::storage::MaskedStorage<components::combat::Speed>>>, &mut specs::storage::Storage<'_, components::digestion::Nutrition, shred::res::Fetch<'_, specs::storage::MaskedStorage<components::digestion::Nutrition>>>, &hibitset::BitSet)` in the current scope
  --> src/systems/combat.rs:83:119
   |
83 |             for (damage, _, speed, mut fullness, _) in (&damages, !&cooldowns, &speeds, &mut fullnesses, &attack_set).join() {
   |                                                                                                                       ^^^^
   |
   = note: the method `join` exists but the following trait bounds were not satisfied:
           `(&specs::storage::Storage<'_, components::combat::Damage, shred::res::Fetch<'_, specs::storage::MaskedStorage<components::combat::Damage>>>, specs::storage::AntiStorage<'_>, &specs::storage::Storage<'_, components::combat::Speed, shred::res::Fetch<'_, specs::storage::MaskedStorage<components::combat::Speed>>>, &mut specs::storage::Storage<'_, components::digestion::Nutrition, shred::res::Fetch<'_, specs::storage::MaskedStorage<components::digestion::Nutrition>>>, &hibitset::BitSet) : specs::join::Join`
```

afaik, I need to implement _some_ trait but I don't know what.

Edit: Okay I fixed the issue now. Not sure if the implementation is correct tho.

**Comment by sunreef** [comment]
You need to add `use amethyst::ecs::Join;` at the top of your file to be able to use the `Join` trait.

**Comment by erlend-sh** [comment]
If you're able to post a gameplay video with your changes applied then I might be able to tell you if you're on the right track.

Anyhow, great that you've started playing with this feature 👏

**Comment by hbina** [comment]
@erlend-sh 

currently, for some reason, the grass won't be consumed...

https://www.youtube.com/watch?v=EW3lU-dHZiY&feature=youtu.be

I think this have to do with the added constraint in the ( ... ).join() I have added in combat.rs causing grasses to be excluded? I am not sure if herbivores are considered as "attackers" of grasses tho.

**Comment by sunreef** [comment]
I think that eventually, we'll have to split attacks and feeding actions into their own distinct systems.

**Comment by hbina** [comment]
@sunreef yep, i did exactly that in my latest commit. do you want me to squash these commits?

https://www.youtube.com/watch?v=DlOHmINbQZA&feature=youtu.be

**Comment by sunreef** [comment]
@hbina I didn't look at the code before commenting. Sorry 😅
We can wait for @marot or @khskarl to give their opinion before we merge.

File: resources/prefabs/creatures/herbivore.ron [hunk 1]
```diff
@@ -31,6 +31,9 @@ Prefab (
                         max: 100.0,
                         value: 100.0,
                     ),
+                    nutrition: (
```
Comment (marotili) [inline resources/prefabs/creatures/herbivore.ron:34]: Could you add nutrition also to plants and herbivores? You can look up the initial values here: https://community.amethyst.rs/t/evoli-mvp-specification/487 (under `Creature attributes`)
Herbivores can't be eaten yet, but we can add it anyway :)

**Review by marotili — APPROVED** [review]
LGTM. Thanks for implementing this and sorry that you had to figure out everything yourself 😅 
We are currently trying to make systems as small as possible. This means systems should have one 'job'. Atm `PerformDefaultAttackSystem` performs three jobs: Applying damage, applying nutrition and starting cooldowns.

The advantage of putting them into separate systems is that the code becomes more configurable and easier to reason about because each system has fewer dependencies.

We will discuss how to best split the system on our [forums](http://community.amethyst.rs), so, for now, your changes are fine :)

File: resources/prefabs/creatures/herbivore.ron (hunk 1 above)
Comment (hbina) [reply resources/prefabs/creatures/herbivore.ron:34]: Added this in https://github.com/amethyst/evoli/pull/69/commits/37f8defcfce7ac4c63764f711dbaf3ccda44b18f

Comment (marotili) [reply resources/prefabs/creatures/herbivore.ron:34]: Great. thanks :)

**Pull request merged**